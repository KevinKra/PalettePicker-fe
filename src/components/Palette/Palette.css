.Palette {
  position: relative;
  border: 2px solid coral;
  border-bottom: 5px solid coral;
  margin: 1.5rem;
  margin-top: 4.5rem;
  height: 100px;
  width: 500px;
  max-width: 25%;
  min-width: 200px;
  display: flex;
  flex-wrap: wrap;
}

.Palette > .color {
  border: 2px solid coral;
  border-top: 3px solid coral;
  border-radius: 5px;
  flex: 1;
  margin: 2%;
  min-width: 5%;
}

.palette-title {
  position: absolute;
  top: -24px;
  left: -8px;
  transform: rotate(-2deg);
  color: rgb(40, 48, 56);
  text-decoration: underline;
}

.palette-delete {
  border-radius: 50px;
  position: absolute;
  font-size: 14px;
  background-color: rgb(40, 48, 56);
  border: 1px solid coral;
  color: white;
  height: 1.6rem;
  width: 1.6rem;
  top: -14px;
  right: -16px;
}

.color {
  cursor: pointer;
}
