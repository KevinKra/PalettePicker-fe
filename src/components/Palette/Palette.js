import React from "react";
import "./Palette.css";

const Palette = props => {
  const colors = props.colors.map((color, i) => {
    return (
      <div
        className="color"
        key={i}
        style={{ backgroundColor: color }}
        onClick={() => console.log("open box")}
      />
    );
  });
  return (
    <article className="Palette">
      <button className="palette-delete">X</button>
      <h4 className="palette-title">{props.title}</h4>
      {colors}
    </article>
  );
};

export default Palette;
