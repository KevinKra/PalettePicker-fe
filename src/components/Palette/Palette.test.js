import React from "react";
import { Palette } from "./Palette";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

describe("Palette", () => {
  let wrapper;
  const historyMock = { push: jest.fn() };
  beforeEach(() => {
    wrapper = shallow(<Palette history={historyMock} colors={["#111"]} />);
  });
  it("should match the snapshot", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it("should invoke the openEditor function when clicked", () => {
    jest;
    const button = wrapper.find(".color");
    button.simulate("click");
    expect(historyMock.push.mock.calls[0]).toEqual(["edit-palette"]);
  });
  xit("should invoke the onClick function when the '.palette-delete' is clicked", () => {
    // UPDATE WITH MERGE
    const button = wrapper.find(".palette-delete");
    button.simulate("click");
    expect(button).toHaveBeenCalled();
  });
});
