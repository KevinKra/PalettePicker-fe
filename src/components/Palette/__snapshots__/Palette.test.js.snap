// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palette should match the snapshot 1`] = `
<article
  className="Palette"
>
  <button
    className="palette-delete"
    onClick={[Function]}
  >
    X
  </button>
  <h4
    className="palette-title"
  />
  <div
    className="color"
    key="0"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#111",
      }
    }
  />
</article>
`;
